libs/network/esp32/jswrap_esp32_network.c

```diff
@@ -92,6 +92,8 @@ static bool g_isStaConnected = false;
 #define EXPECT_CB_EXCEPTION(jsCB)   jsExceptionHere(JSET_ERROR, "Expecting callback function but got %v", jsCB)
 #define EXPECT_OPT_EXCEPTION(jsOPT) jsExceptionHere(JSET_ERROR, "Expecting Object, got %t", jsOPT)
 
+// Global data structure for setIP and setAPIP
+tcpip_adapter_ip_info_t info;
```

Comment:
Please could you move this into the setIP function? Looking at tcpip_adapter_set_ip_info I'm pretty sure it doesn't have to be a global variable so having it just in the fn will free up a bit of RAM